### PR TITLE
fix #94 external gps update

### LINKED ISSUE #94: Update GPS external connection
existing documentatiton is misleading after merging https://github.com/lutraconsulting/input/pull/1886 and related

https://github.com/MerginMaps/docs/blob/main/src/field/external_gps.md

### PR DESCRIPTION
**Pull request by alex-cit**
fix #94 
- update of external GPS use in Android/ iOS
- some troubleshooting tips
- List of known supported GPS devices

I feel like we should (could) introduce the term GNSS here, e.g., when talking about GNSS corrections.

### CODE REVIEW

File: src/field/external_gps.md [hunk 1]
```diff
@@ -1,44 +1,66 @@
...
+![input settings](./input-external-gps.png)
+3. Tap on **Connect new receiver** in GPS receivers panel:
+![input-connect GPS](./external-gps-connect.png)
+4. Choose your GPS receiver from the list of Bluetooth devices. You might be asked to pair your device during this process.
+![input GPS receivers](./external-gps-connect2.png)
+5. You can now switch between internal and external receivers:
+![input GPS receivers](./external-gps-receivers.png)
 
-## Android
+### How to connect external GPS receiver in Android
```
Comment (PeterPetrik) [inline src/field/external_gps.md:29]: Maybe: How to connect external GPS receiver in Android (mock)

File: src/field/external_gps.md [hunk 2]
```diff
@@ -1,44 +1,66 @@
...
+- Make sure that your mobile device offers the functionality to pair with an external GPS device and that it communicates through a standard interface.
 
-For example, earlier Android devices did not offer system level GPS override from an external device.
+## External GPS in Android
+In Android, there are two ways how to use an external GPS device in Input:
+- directly through Input as an external receiver
+- connecting GPS receiver to your mobile device using an app that will provide mock location to Input
 
-In this section, we explore connecting to an external GPS device through Bluetooth using Android and iOS mobiles/tablets.
+### How to connect external GPS receiver in Input app
```
Comment (PeterPetrik) [inline src/field/external_gps.md:18]: Maybe: How to connect external GPS receiver in Android (directly)

File: src/field/external_gps.md [hunk 3]
```diff
@@ -1,44 +1,66 @@
...
+- In Android device, enable **Developer options**, which are hidden by default:
+  - On newer Android versions, go to **Android Settings** > **About phone** and scroll down to find the **Build number**. Tap the build number 7 times. A message appears with a count-down until you **become a developer**.
+  - On older Android versions, the developer settings can be made visible under the **Android Settings** > **Applications** > **Development**.
+- Once you have the developers option enabled, go to **System** > **Advanced** > **Developer options** > **Select mock location app** and select your GPS app.
 
-![Mcok location](./external_gps_android_mock.png)
+The mock location will be used by Input automatically as if received from the internal receiver.
 
-Now, pair your external GPS device to your mobile/tablet through Bluetooth. Then open the **Bluetooth GPS** app and from the **Select paired GPS device and connect** drop-down menu, select your external GPS. Make sure you have also selected the box for **Enable Mock GPS Provider**. Select **Connect**. You can now open Input and it should be using the external GPS device.
+## External GPS in iOS
```
Comment (PeterPetrik) [inline src/field/external_gps.md:46]: I am not sure, but isn't correct to say "**on** iOS, **on** Android"?

File: src/field/external_gps.md [hunk 4]
```diff
@@ -1,44 +1,66 @@
...
 
-![GPS Bluetooth](./external_gps_android_bluetooth.png)
+External GPS can be connected to your iOS device via Bluetooth. Depending on the GPS manufacturer, there may be a custom app, which then provides position to your iOS device. Input sees this as an internal GPS receiver automatically, without additional configuration.
 
+![GPS iOS](./external-gps-ios.png)
 
-## iOS
-To use your external GPS device with iPhone or iPad, you need to pair the device via Bluetooth. Once the device is paired, the signal from external GPS will be used as the primary source.
+## External GPS troubleshooting
+If you encounter issues with external GPS in Input, make sure that:
```
Comment (PeterPetrik) [inline src/field/external_gps.md:53]: Input -> use component

File: src/field/external_gps.md [hunk 5]
```diff
@@ -1,44 +1,66 @@
...
+- your Bluetooth is turned on
+- GPS is paired with your mobile device
+- your GPS receiver is turned on, has battery and is sending data
+- remove the device in Input, and add it again from Input
```
Comment (PeterPetrik) [inline src/field/external_gps.md:57]: Input -> use component

**Review by PeterPetrik — COMMENTED** [review]
Looks good to me

@tomasMizera can you please look too?

**Comment by PeterPetrik** [comment]
Unable to merge as it is marked as draft

**Comment by PeterPetrik** [comment]
@saberraz if I remember you had some external GPS too that you test against? which model?

**Comment by saberraz** [comment]
Mine is this one: https://www.garmin.com/en-GB/p/645104

**Comment by PeterPetrik** [comment]
In the list so far: **Emlid Reach RS+**,  **Leica FLX100**, **Trimble R2**

File: src/field/external_gps.md [hunk 6]
```diff
@@ -1,44 +1,66 @@
...
 
-::: warning
-Users have reported that they have to switch on Aeroplane mode to be able to use the external GPS device.
-:::
+## List of known supported GPS devices
+External GPS functionality depends on the manufacturer and on the specific model of your GPS device. These devices are currently known to work well:
+
+- **Emlid Reach RS+** - directly via Bluetooth connection, has an internal NTRIP client to receive corrections. Android only.
```
Comment (PeterPetrik) [inline src/field/external_gps.md:62]: We can also have a markdown table , and this text below table as extra details.

- Manufacturer | Model | Android | iOS | 
- XX.                  | YYYY  | yes/no/unknown | yes/no/unknown |

**Review by PeterPetrik — COMMENTED** [review]
looks good

**Comment by PeterPetrik** [comment]
> Mine is this one: https://www.garmin.com/en-GB/p/645104

@saberraz iOS/Android/both?

**Comment by saberraz** [comment]
> > Mine is this one: https://www.garmin.com/en-GB/p/645104
> 
> @saberraz iOS/Android/both?

Only Android. iOS is a blackbox when it comes to Bluetooth!

**Comment by saberraz** [comment]
See also:
https://www.linkedin.com/feed/update/urn:li:ugcPost:6902893802277978112?commentUrn=urn%3Ali%3Acomment%3A%28ugcPost%3A6902893802277978112%2C6905838597531144192%29

**Review by tomasMizera — COMMENTED** [review]
Good job @alex-cit

My main concern is that we should tell users that mock location is not a recommended way even though for some devices on Android it is the only way.

See discussion here regarding GPS vs GNSS naming: https://lutraconsulting.slack.com/archives/CC95JK1T9/p1636449250100900
If we stick with GPS, we could add a note somewhere that we are capable of reading also GLONASS, Galileo,.. so that they know it is just a naming

**Comment by PeterPetrik** [comment]
@alex-cit 

[Simon Anderson](https://merginmaps.slack.com/archives/C02DWECUEN8/p1646952750995389) on slack:

"
Geode GNS3 - through Geode Connect app
(https://play.google.com/store/apps/details?id=com.juniper.geode2a&hl=en_NZ&gl=US)
(https://apps.apple.com/us/app/geode-connect/id1446098695)
which also acts as an NTRIP client and sends corrections to the device.
Manufacturer | Model | Android | iOS |
Juniper Systems | GNS3 | yes | yes |
I feel like we should (could) introduce the term GNSS here, e.g., when talking about GNSS corrections . - Some of these receivers are available as GPS only or GNSS
"

**Review by PeterPetrik — APPROVED** [review]

**Pull request merged**